rpcs3/Emu/Cell/lv2/sys_game.cpp

```diff
@@ -7,11 +7,11 @@
 
 LOG_CHANNEL(sys_game);
 
-error_code _sys_game_board_storage_read(vm::ptr<u8> buffer, u8 code)
+error_code _sys_game_board_storage_read(vm::ptr<u8> buffer1, vm::ptr<u8> buffer2)
 {
-	sys_game.trace("sys_game_board_storage_read(buffer=*0x%x, code=0x%02x)", buffer, code);
+	sys_game.trace("sys_game_board_storage_read(buffer1=*0x%x, buffer2=*0x%x)", buffer1, buffer2);
 
-	if (!buffer)
+	if (!buffer1 || !buffer2)
```

Comment:
Second EFAULT check is checked after buffer1 write, the way most EFAULT checks works is it happensjust before the write.

Reply:
Moved the EFAULT check for buffer2.